service/history/events/cache.go

```diff
@@ -83,6 +85,27 @@ var (
 
 var _ Cache = (*cacheImpl)(nil)
 
+// NewEventCache creates a new global events cache
+func NewEventCache(
```

Comment:
Maybe rename to `NewGlobalCache`? The function is already in `events` package, so `NewEventsCache` sounds a little bit redundant.